operation/web.go

```diff
@@ -0,0 +1,118 @@
+package operation
+
+import (
+	"context"
+
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+const opidMetadataKey = "opid"
+
+// UnaryClientInterceptor adds the operation id from the current context (if any) to the
+// outgoing unary RPC metadata.
+func UnaryClientInterceptor(
+	ctx context.Context,
+	method string,
+	req, reply interface{},
+	cc *grpc.ClientConn,
+	invoker grpc.UnaryInvoker,
+	opts ...grpc.CallOption,
+) error {
+	if op := Get(ctx); op != nil && op.ID.String() != "" {
+		ctx = metadata.AppendToOutgoingContext(ctx, opidMetadataKey, op.ID.String())
+	}
+	return invoker(ctx, method, req, reply, cc, opts...)
+}
+
+// StreamClientInterceptor adds the operation id from the current context (if any) to the
+// outgoing streaming RPC metadata.
+func StreamClientInterceptor(
+	ctx context.Context,
+	desc *grpc.StreamDesc,
+	cc *grpc.ClientConn,
+	method string,
+	streamer grpc.Streamer,
+	opts ...grpc.CallOption,
+) (grpc.ClientStream, error) {
+	if op := Get(ctx); op != nil && op.ID.String() != "" {
+		ctx = metadata.AppendToOutgoingContext(ctx, opidMetadataKey, op.ID.String())
+	}
+	return streamer(ctx, desc, cc, method, opts...)
+}
+
+// UnaryServerInterceptor creates a new operation in the current context before passing
+// it to the unary response handler. If the incoming RPC metadata contains an operation
+// ID, the new operation will have the same ID.
+func (m *Manager) UnaryServerInterceptor(
```

Comment:
I know that we have the method as part of the manager struct because of the function signature, but it feels weird to me - I don't think passing the manager in would be too bad

Reply:
i'm also not too happy with this, but I think passing in a manager is a similarly awkward and results in more boilerplate code. lemme think about a nicer way to structure this.

Reply:
I'm leaning towards leaving this as is - the operation manager has other methods that assume a request and response.  Longer-term I think we can decouple the web-layer logic from the `operation` package and the manager struct.

Reply:
sgtm